src/test/components/DragAndDrop.test.js

```diff
@@ -59,4 +77,65 @@ describe('app/DragAndDrop', () => {
     fireEvent.dragExit(dragAndDrop);
     expect(overlay.classList).not.toContain('dragging');
   });
+
+  it('receives profile on file drop', async () => {
+    const { container } = render(
+      <Provider store={createStore()}>
+        <DragAndDrop>Target area here</DragAndDrop>
+      </Provider>
+    );
+    const [dragAndDrop, overlay] = container.children;
+
+    function mockFile({ type, payload }): File {
+      const file = {
+        type,
+        _payload: payload,
+      };
+      return (file: any);
+    }
+    /**
+     * Bypass all of Flow's checks, and mock out the file reader.
+     */
+    function mockFileReader(mockFile: File) {
+      const payload = (mockFile: any)._payload;
+      return {
+        asText: () => Promise.resolve((payload: string)),
+        asArrayBuffer: () => Promise.resolve((payload: ArrayBuffer)),
+      };
+    }
+    const store = blankStore();
+
+    const setupTestWithFile = async mockFileOptions => {
+      const file = mockFile(mockFileOptions);
+      const { dispatch } = store;
+      const views = (
+        await observeStoreStateChanges(store, () =>
```

Comment:
It's probably not needed to observe all store changes. Instead we can wait until everything is loaded.
Some possibilities:
1. (my preference, but could require more setup) put an existing component inside DragAndDrop, that reacts on this state, for example `AppViewRouter`. Then you can use the testing library functions such as `findByText` that wait until they find the text in the DOM.
2. use react-testing-library's `waitFor` function to "wait for" some state is right. ([see documentation](https://testing-library.com/docs/dom-testing-library/api-async#waitfor)). (important information: `waitFor` waits until the function doesn't throw -- it doesn't wait for a boolean result. So you should use `expect` if you use this path).

My preference is 1 but I would be happy enough with 2 which may be easier for you, so maybe try to do 2 first, and then we can look at doing 1 if you want to.